hypothesis-python/src/hypothesis/core.py

```diff
@@ -160,8 +160,62 @@ class Example:
     reason: Any = field(default=None)
 
 
+# TODO_DOCS link to not-yet-existent patch-dumping docs
+
+
 class example:
-    """A decorator which ensures a specific example is always tested."""
+    """
+    Add an explicit input to a Hypothesis test, which Hypothesis will always
+    try before generating random inputs. This combines the randomized nature of
+    Hypothesis generation with a traditional parametrized test.
+
+    For example:
+
+    .. code-block:: python
+
+        @example("Hello world")
+        @example("some string with special significance")
+        @given(st.text())
+        def test_strings(s):
+            pass
+
+    will first call ``test_strings("Hello World")``, then
+    ``test_strings("some string with special significance")``, and then generate
+    inputs randomly.
+
+    Explicit inputs from |@example| are run in the |Phase.explicit| phase.
+    Explicit inputs do not count towards |settings.max_examples|.
+
+    |@example| can also be used to easily reproduce a failure. For instance, if
+    Hypothesis reports that ``f(n=[0, math.nan])`` fails, you can add
+    ``@example(n=[0, math.nan])`` to your test to quickly reproduce that failure.
+
+    Note that explicit inputs added by |@example| do not shrink. If an
+    explicit input fails, Hypothesis will stop immediately and report that
+    failure.
```

Comment:
I'd concat this with the phase + max_examples paragraph.

It's also not strictly true that we stop immediately; with report_multiple_failures we'll run all explicit inputs and _then_ stop.  I'd instead say something like "... Hypothesis will stop and report the failure without generating any random inputs".